core/creator/src/main/java/io/quarkus/creator/phase/nativeimage/NativeImagePhase.java

```diff
@@ -345,12 +346,16 @@ public void provideOutcome(AppCreator ctx) throws AppCreatorException {
                     throw new AppCreatorException("GRAALVM_HOME was not set");
                 }
             }
-            nativeImage = Collections.singletonList(graalvmHome + File.separator + "bin" + File.separator + "native-image");
+            if (IS_WINDOWS) {
```

Comment:
Have you considered a condensed version?

Something like:
```
String nativeCommand = IS_WINDOWS ? "native-image.cmd" : "native-image";
nativeImage = Collections.singletonList(graalvmHome + File.separator + "bin" + File.separator + nativeCommand);
```

Reply:
Sure, I just have a feeling that Windows related changes are just about add up going forward.

But I will update as suggested.

Reply:
changed